### PR TITLE
Rework dockercfg in builder package to use upstream keyring

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
(pending testing)

[test]

**Comment by mfojtik** [comment]
Fixes #2624 

@deads2k FYI

File: pkg/build/builder/cmd/dockercfg/cfg.go [hunk 1]
```diff
@@ -39,8 +40,7 @@ func (h *Helper) InstallFlags(flags *pflag.FlagSet) {
 
 // GetDockerAuth returns a valid Docker AuthConfiguration entry, and whether it was read
 // from the local dockercfg file
-func (h *Helper) GetDockerAuth(registry, authType string) (docker.AuthConfiguration, bool) {
-	var authCfg docker.AuthConfiguration
+func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfiguration, bool) {
```
Comment (deads2k) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: `registry` to `imageName`?  I suspect this change in meaning need to rippled back to callers?

Comment (deads2k) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: Also, while we're here, why a `bool` instead of an `error`?

Comment (mfojtik) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: oh right, I should not be doing multiple PR's in same time... :-) will fix that

**Comment by mfojtik** [comment]
@deads2k just realized we don't have any integration test-cases for this so I will have to check this manually...

**Comment by deads2k** [comment]
or write an integration test..... :)

**Comment by mfojtik** [comment]
@deads2k I tested this locally with `pullSecret` specified manually and it works. can you have second look?

File: pkg/build/builder/cmd/dockercfg/cfg.go [hunk 2]
```diff
@@ -48,31 +48,21 @@ func (h *Helper) GetDockerAuth(registry, authType string) (docker.AuthConfigurat
...
 	if _, err := os.Stat(dockercfgPath); err != nil {
 		glog.V(3).Infof("Problem accessing %s: %v", dockercfgPath, err)
-		return authCfg, false
+		return docker.AuthConfiguration{}, false
 	}
 	cfg, err := readDockercfg(dockercfgPath)
 	if err != nil {
 		glog.Errorf("Reading %s failed: ", dockercfgPath, err)
-		return authCfg, false
+		return docker.AuthConfiguration{}, false
```
Comment (deads2k) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: I'm still not completely convinced that this method shouldn't return an error for cases like this.  This looks like a failure to fulfill intent.

**Comment by deads2k** [comment]
One optional comment.  Otherwise, lgtm.

File: pkg/build/builder/cmd/dockercfg/cfg.go (hunk 2 above)
Comment (mfojtik) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: @deads2k it is not error in context of this function as when the dockercfg does not exists or the reading of it fails it means we have no dockercfg and we skip it (and log the error to build log). We can come to review this post-3.0 (i can create follow up trello card)

Comment (deads2k) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: > it is not error in context of this function as when the dockercfg does not exists or the reading of it fails it means we have no dockercfg and we skip it (and log the error to build log). We can come to review this post-3.0 (i can create follow up trello card)

ok.  [merge]

**Comment by deads2k** [comment]
ah, needs rebase.

**Comment by mfojtik** [comment]
@deads2k rebased [merge]

**Comment by mfojtik** [comment]
@deads2k I think we have another flaky test.... I tried to run test-cmd locally and it works. Will wait for merge and if that fails I will have to go back investigate

**Pull request merged**